Pocket/pocket-ios
# Tags View: fix some issues

## Summary
* This PR fixes a few issues with the Tags Filter View:
    - Edit button not working on iOS 17+
    - Rename dialog does not have a cancel button, forcing to make a change if tapped by mistake
* Some refactors of the views was also done

> [!NOTE]
> the edit button had issues on iOS 17, according to Apple fixed in iOS 17.2. If you test on iOS 17 prior than 17.2 you might see issues with the edit button (e.g. POCKET-9440)

## Implementation Details
* Refactored `TagsFilterView` and related subviews
## Test Steps
* Go to Saves->Tagged
* Make sure `Edit/Done` work properly
* Perform some rename/delete operartions make sure they work
* Make sure you can always enter/exit the edit mode

## PR Checklist:

## Screenshots
